spec/presenters/payroll_data_csv_row_spec.rb

```diff
@@ -0,0 +1,61 @@
+require "rails_helper"
+
+RSpec.describe PayrollDataCsvRow do
+  subject { described_class.new(claim) }
+  let(:claim) { build(:claim) }
+
+  describe "to_s" do
+    let(:row) { CSV.parse(subject.to_s).first }
+    let(:start_of_month) { Date.today.at_beginning_of_month }
+
+    let(:claim) do
+      build(:claim, :submittable,
+        payroll_gender: :female,
+        date_of_birth: Date.new(1980, 12, 1),
+        student_loan_plan: StudentLoans::PLAN_2,
+        bank_sort_code: "001122",
+        bank_account_number: "01234567",
+        eligibility: build(:student_loans_eligibility, :eligible))
+    end
+
+    it "generates a csv row" do
+      expect(row).to eq([
+        nil,
+        claim.first_name,
+        claim.middle_name,
+        claim.surname,
+        claim.national_insurance_number,
+        "F",
+        start_of_month.strftime("%m/%d/%Y"),
+        (start_of_month + 7.days).strftime("%m/%d/%Y"),
+        claim.date_of_birth.strftime("%m/%d/%Y"),
+        claim.email_address,
+        claim.address_line_1,
+        claim.address_line_2,
+        claim.address_line_3,
+        claim.address_line_4,
+        nil,
+        claim.postcode,
+        "United Kingdom",
+        "BR",
+        "0",
+        "3",
+        "A",
+        "T",
+        "2",
+        claim.full_name,
+        claim.bank_sort_code,
+        claim.bank_account_number,
+        "Scheme B",
+        claim.eligibility.student_loan_repayment_amount.to_s,
+        claim.reference,
+      ])
+    end
+
+    it "escapes fields with strings that could be dangerous in Microsoft Excel and friends" do
+      claim.address_line_1 = "=ActiveCell.Row-1,14"
+
+      expect(row[PayrollDataCsv::FIELDS.index(:address_line_1)]).to eq("\\#{claim.address_line_1}")
```

Comment:
I'm suspiscious of this. Do backslashes actually escape equals signs in Excel?

Reply:
I'm not sure. When I went through it with Tamas, he seemed happy enough with prepending these with a backslash

Reply:
Could we test it? When I was researching this, I was put under the impression that escaping was more involved.

Reply:
Yeah, I'll find a way to test it 👍

Reply:
more details on the current approach to escaping in the history – https://github.com/dFE-Digital/dfe-teachers-payment-service/commit/eda1b969

Reply:
My question is: does `\` actually escape characters in spreadsheet software? The article linked in that commit suggests the only reliable way of escaping formulae is with a prefixed tab character.

Reply:
hmmm, I seemed pretty confident it would in my commit message. I did test it in Excel, Google Sheets and Numbers, and it prevented the formula being evaluated in all of those. I think the reason the article linked to in in the message suggests a tab character is because the "tab character doesn’t show up visually on Excel, or Google Sheets", and therefor it's a method that doesn't result in the data looking weird. In our case we are being a bit more blunt and visible with the `\`

Reply:
Ok, cool. Just wanted to make sure. My research suggested that wasn't an option (or rather everyone goes to more extreme lengths to get it done). Is it ok that we have `\`s in the data we send to Cantium?

Reply:
I'd say the likelyhood of that is fairly low. For it to happen, a user would have to enter highly unusual data in an address field (whether malicious or otherwise) and the claim would then need to be approved. In that situation some manual intervention may be required (either by them or us) if the presence of a backslash is going to cause a problem. On balance we are probably best crossing that edge-case if/when it arises to remove the security risk.